### PR TITLE
MINOR: Use self-managed mode instead of KIP-500 and nozk

### PR DESCRIPTION
**Pull request by ijuma**
KIP-500 is not particularly descriptive. I also tweaked the readme text a bit.

Tested that the readme for self-managed still works after these changes.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -663,19 +663,18 @@ object KafkaConfig {
...
   val NodeIdDoc = "The node ID associated with the roles this process is playing when `process.roles` is non-empty. " +
     "This is required configuration when the self-managed quorum is enabled."
-  val MetadataLogDirDoc = "This configuration determines where we put the metadata log for clusters upgraded to " +
-    "KIP-500. If it is not set, the metadata log is placed in the first log directory from log.dirs."
-  val ControllerListenerNamesDoc = "A comma-separated list of the names of the listeners used by the KIP-500 controller. This is required " +
-    "if this process is a KIP-500 controller. The ZK-based controller will not use this configuration."
+  val MetadataLogDirDoc = "This configuration determines where we put the metadata log for clusters in self-managed mode. " +
+    "If it is not set, the metadata log is placed in the first log directory from log.dirs."
+  val ControllerListenerNamesDoc = "A comma-separated list of the names of the listeners used by the self-managed controller. This is required " +
+    "if this process is a self-managed controller. The ZK-based controller will not use this configuration."
```
Comment (rondagostino) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: > This is required if this process is a self-managed controller.

This is actually required for self-managed brokers as well -- it is how the broker figures out what security protocol to use when connecting to the controller.quorum.voters.

Maybe `s/if this process is a self-managed controller/if this process is a self-managed broker and/or controller/`

**Review by rondagostino — COMMENTED** [review]
I agree with these changes.  I noted one incorrect statement in a config documentation string that pre-dated these changes and needs updating here.

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (cmccabe) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: How about "this is required if this process is part of a self-managed cluster"

**Comment by cmccabe** [comment]
This all seems reasonable.  It's nice to have a link from the main readme.

One nitpick: If we're going to have a subdirectory for the self-managed configurations, then we don't need to also have "self-managed" in the file name, I think.  For example, `./config/self-managed/self-managed-combined.properties` can just be `./config/self-managed/combined.properties`, etc.

**Comment by hachikuji** [comment]
I agree with Colin that we should drop the "self-managed" prefix from the files under the directory with the same name.

**Comment by ijuma** [comment]
@cmccabe @hachikuji I considered removing the prefix from the file, but I thought people may copy the files to other locations and the file name should be descriptive. If you both still feel it should be renamed, I am happy to do it though. Let me know.

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (ijuma) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: Applied Colin's change.

**Comment by cmccabe** [comment]
I don't have a strong opinion but I think on balance I'd prefer to remove the prefix

**Comment by ijuma** [comment]
I renamed the files, made a few tweaks to the readme and went through it to make sure it still works. Really cool btw, exciting to see all of this working. :)

**Review by cmccabe — APPROVED** [review]
LGTM pending jenkins

**Comment by ijuma** [comment]
Unrelated flaky failure:

> RebalanceSourceConnectorsIntegrationTest.testMultipleWorkersRejoining

**Comment by ijuma** [comment]
Merged to trunk and cherry-picked to 2.8.

**Pull request merged**